src/compiler/parse_table.cc

```diff
@@ -151,7 +151,7 @@ bool ParseState::has_shift_action() const {
 void ParseState::each_referenced_state(function<void(ParseStateId *)> fn) {
   for (auto &entry : terminal_entries)
     for (ParseAction &action : entry.second.actions)
-      if (action.type == ParseActionTypeShift || ParseActionTypeRecover)
+      if (action.type == ParseActionTypeShift || action.type == ParseActionTypeRecover)
```

Comment:
😬  Oh wow. Thanks for catching that.

We need to check one other condition here to ensure that `action` will have a meaningful `state_index` field:

```cpp
if ((action.type == ParseActionTypeShift && !action.extra) || action.type == ParseActionTypeRecover)
  fn(&action.state_index)
```

**Background** - There's a special 'shift-extra' action that tells the parser to consume an 'extra token' (e.g. a comment). These actions don't change the parser's state, so they don't have a `state_index` field. I can't remember why I modeled these actions using a separate boolean `extra` field; it seems like there should just be a separate `ParseActionTypeShiftExtra` value for the `type` field. But anyway, with the current setup, only shift actions that are *not* `extra` will have a meaningful `state_index`.